remote/remote.go

```diff
@@ -235,12 +277,19 @@ func (r *Remote) GoOnline(ctx context.Context) error {
 // the dataset ref from the refstore and add the (n + 1)th to the refstore
 // gen = -1 should indicate that we remove all the dataset versions
 func (r *Remote) RemoveDataset(ctx context.Context, params map[string]string) error {
-	pid, ref, err := r.pidAndRefFromMeta(params)
+	subj, ref, err := r.subjAndRefFromMeta(params)
 	if err != nil {
 		return err
 	}
 	log.Debugf("remove dataset %s", ref)
 
+	pid := subj.ID
+	if r.policy != nil {
+		if err := r.policy.Enforce(subj, strings.Join([]string{"dataset", ref.Username, ref.Name}, ":"), "remote:remove"); err != nil {
```

Comment:
I'd pull the construction of `dataset:username:name` into a variable one line up & pass the var here. There's a very high chance we'll define helper functions in the `access` package for this.

Apply this advice to all hooks

Reply:
I created a helper function in `access` for this purpose. Used it in `logPreCheckHook` and nowhere else 🤦‍♀️